? PATCHES.txt
? translation_management_remove_popup.patch
? translation_management_set_to_finished_if_empty_field_data.patch
? translation_management_wysiwyg_support_for_cck_textarea.patch
Index: icl_translate/icl_translate.editor_form.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/translation_management/icl_translate/icl_translate.editor_form.inc,v
retrieving revision 1.55.2.6
diff -u -p -r1.55.2.6 icl_translate.editor_form.inc
--- icl_translate/icl_translate.editor_form.inc	17 Dec 2010 04:01:47 -0000	1.55.2.6
+++ icl_translate/icl_translate.editor_form.inc	19 Jan 2011 13:21:18 -0000
@@ -133,7 +133,7 @@ function icl_translate_editor($form) {
       $form[$field['field_type']] = array (
         '#type' => 'fieldset', 
         '#title' => ucfirst(str_replace('_', ' ', t($title))),
-        '#collapsible' => TRUE
+        '#collapsible' => TRUE,
       );
       
       // Check if doesn't need update, compare previous requests
@@ -174,24 +174,62 @@ function icl_translate_editor($form) {
         '#name' => $name . '[field_finished]',
         '#default_value' => $finished
       );
-      
-      // Set translation content field
       $element_type = _icl_translate_editor_get_element_type($field['field_type']);
-      $lenght = strlen(strip_tags($field_data_translated));
-      $rows = round($lenght/60) + 2;
-      $form[$field['field_type']]['field_data_translated']['field_data_translated'] = array(
-        '#title' => sprintf(t('Translated content - %s'), $target_lang['icl_name']),
-        '#name'=> $name . '[field_data_translated]',
-        '#type' => $element_type,
-        '#default_value' => $field_data_translated,
-        '#rows' => $rows,
-        '#id' => 'edit-field-data-translated-' . _icl_translate_editor_sanitize_id($field['field_type']) . '-' . $type['element_id'],
-      );
-      
-      if (user_access('administer filters') && $type['type'] == 'node' && $field['field_type'] == 'body') {
+      // Set translation content field
+      if (is_array($field_data_translated)) {
+        $form[$field['field_type']]['field_data_translated'] = array(
+          '#name' => 'field_data_translated',
+          '#tree' => TRUE,
+        );
+        foreach ($field_data_translated as $key => $translated_field) {
+          $length = strlen(strip_tags($field_data_translated));
+          $rows = round($length / 60) + 2;
+          $form[$field['field_type']]['field_data_translated'][$key] = array(
+            '#title' => sprintf(t('Translated content - %s'), $target_lang['icl_name']),
+            '#name' => $name . '[field_data_translated][' . $key . ']',
+             '#type' => $element_type,
+             '#default_value' => $translated_field,
+             '#rows' => $rows,
+             '#id' => 'edit-field-data-translated-' . _icl_translate_editor_sanitize_id($field['field_type']) . '-' . $type['element_id'],
+          );
+          preg_match("/^(CCK)\[([a-z_A-Z0-9]*)\]/", $field['field_type'], $matches);
+          if ($matches[1] == 'CCK') {
+            $cck_field = content_fields($matches[2]);
+             if ($cck_field['widget']['type'] == 'text_textarea'){
+               $format = $cck_field['widget']['default_value'][$key]['format'];
+               $cck_area = true;
+             }
+          }
+          if ($type['type'] == 'node' && ($field['field_type'] == 'body'  || $cck_area)) {
+            $form[$field['field_type']]['field_data_translated']['format'] = filter_form($format, NULL, array($field['field_type'], 'field_data_translated_format'));
+          }
+        }
+      }
+      else {
+        $length = strlen(strip_tags($field_data_translated));
+        $rows = round($length / 60) + 2;
+        $form[$field['field_type']]['field_data_translated']['field_data_translated'] = array(
+            '#title' => sprintf(t('Translated content - %s'), $target_lang['icl_name']),
+            '#name' => $name . '[field_data_translated]',
+            '#type' => $element_type,
+            '#default_value' => $field_data_translated,
+            '#rows' => $rows,
+            '#id' => 'edit-field-data-translated-' . _icl_translate_editor_sanitize_id($field['field_type']) . '-' . $type['element_id'],
+        );
+        
+        preg_match("/^(CCK)\[([a-z_A-Z0-9]*)\]/", $field['field_type'], $matches);
+        if ($matches[1] == 'CCK') {
+          $cck_field = content_fields($matches[2]);
+          if ($cck_field['widget']['type'] == 'text_textarea'){
+               $format = $cck_field['widget']['default_value'][0]['format'];
+               $cck_area = true;
+             }
+        }
         $form[$field['field_type']]['field_data_translated']['format'] = filter_form($format, NULL, array($field['field_type'], 'field_data_translated_format'));
       }
-      
+      if (is_array($field_data)) {
+        $field_data = implode('<br />', $field_data);
+      }
       // Set original content field
       $form[$field['field_type']]['field_data'] = _icl_wrapper_form_create_markup(array(
         '#value' => '<div class="icl_translate_hidden original_data">'
@@ -250,7 +288,7 @@ function icl_translate_editor_submit(&$f
   if (isset($_POST['rid'])) {
     $rid = $_POST['rid'];
   }
-  
+
   if (!isset($_POST['finished'])) {
     _icl_wrapper_form_create_redirect($form, array('translator'));
   }
@@ -275,15 +313,12 @@ function icl_translate_editor_submit(&$f
       
       // Process CCK
       if ($type == 'CCK') {
-        
         foreach ($data as $cck => $cck_data) {
           $cck_data['field_finished'] = isset($cck_data['field_finished']) ? 1 : 0;
-          $cck_update = explode(',', $cck_data['field_data_translated']);
-          
-          foreach ($cck_update as $cck_update_key => $cck_update_value) {
-            $cck_update[$cck_update_key] = trim($cck_update_value);
+          $cck_update = $cck_data['field_data_translated'];
+          foreach ($cck_update as &$cck_update_value) {
+            $cck_update_value = trim($cck_update_value);
           }
-          
           // Autosave
           if (isset($_REQUEST['icl_ajax'])) {
             _icl_translate_editor_autosave($rid, 'CCK[' . $cck . ']', $cck_update, $cck_data['field_finished']);
@@ -405,11 +440,6 @@ function _icl_translate_editor_sanitize_
     
     default:
       $data = unserialize($data);
-      if (!empty($data)) {
-        if (is_array($data)) {
-          $data = implode(', ', array_values($data));
-        }
-      }
   }
   return $data;
 }
@@ -794,4 +824,4 @@ function _icl_translate_editor_get_eleme
   }
   
   return $element_type;
-}
\ No newline at end of file
+}
